concent_api/core/views.py

```diff
@@ -373,6 +406,100 @@ def handle_send_force_get_task_result(request, client_message: message.concents.
+
+    if Message.objects.filter(
+        type    = client_message.TYPE,
+        task_id = client_message.ack_report_computed_task.task_to_compute.compute_task_def['task_id']
+    ).exists():
+        return message.concents.ServiceRefused(
+            header = MessageHeader(
+                type_     = message.concents.ServiceRefused.TYPE,
+                timestamp = current_time,
+                encrypted = False,
+            ),
+            reason      = message.concents.ServiceRefused.REASON.DuplicateRequest,
+        )
+    if not is_provider_account_status_positive(client_message) or tmp_is_provider_account_status_positive(request):
+        return message.concents.ServiceRefused(
+            header = MessageHeader(
+                type_     = message.concents.ServiceRefused.TYPE,
+                timestamp = current_time,
+                encrypted = False,
+            ),
+            reason      = message.concents.ServiceRefused.REASON.TooSmallProviderDeposit,
+        )
+    client_message_send_to_soon = client_message.ack_report_computed_task.timestamp + settings.CONCENT_MESSAGING_TIME
+    client_message_send_to_late = client_message.ack_report_computed_task.timestamp + settings.CONCENT_MESSAGING_TIME + settings.FORCE_ACCEPTANCE_TIME
+    if client_message_send_to_late < current_time:
+        return message.concents.ForceSubtaskResultsRejected(
+            header = MessageHeader(
+                type_     = message.concents.ForceSubtaskResultsRejected.TYPE,
+                timestamp = current_time,
+                encrypted = False,
+            ),
+            reason = message.concents.ForceSubtaskResultsRejected.REASON.RequestTooLate,
+        )
+    elif current_time > client_message_send_to_soon:
+        return message.concents.ForceSubtaskResultsRejected(
+            header = MessageHeader(
+                type_     = message.concents.ForceSubtaskResultsRejected.TYPE,
+                timestamp = current_time,
+                encrypted = False,
+            ),
+            reason = message.concents.ForceSubtaskResultsRejected.REASON.RequestPremature,
+        )
+    else:
+        client_message.sig = None
+        store_message_and_message_status(
+            client_message.TYPE,
+            client_message.ack_report_computed_task.task_to_compute.compute_task_def['task_id'],
+            client_message.serialize(),
+            status               = ReceiveStatus,
+            provider_public_key  = client_public_key,
+            requestor_public_key = other_party_public_key,
+        )
+        return HttpResponse("", status = 202)
+
+
+def handle_send_force_subtask_results_results_response(request, client_message):
+    assert client_message.TYPE in message.registered_message_types
+
+    current_time      = int(datetime.datetime.now().timestamp())
+    client_public_key = decode_client_public_key(request)
+
+    if isinstance(client_message.subtask_results_accepted, message.tasks.SubtaskResultsAccepted):
```

Comment:
`assert` should be added here that `client_message.subtask_results_accepted` contains either `subtask_results_accepted` or `subtask_results_rejected`.